### DISCUSSION TITLE
mkcert challenge

### DISCUSSION THREAD

**Discussion by vpermits** [ddev/ddev#4621] [General]
php 8.1, Drupal 9.4.10
mkcert shows:
The local CA is already installed in the system trust store! 
The local CA is already installed in the Firefox trust store! 
The local CA is already installed in Java's trust store! 
I have two ddev projects running on local dev machine and they work great.
I'm trying to make an API call from https on machine 1 to https on machine 2 hitting /oauth/token using curl_exec with good options and call is returning false with exception:
SSL: no alternative certificate subject name matches target host name 'united-road-1.ddev.site'

the ssl cert on machine one is valid with https://ipm.ddev.site

Even tried:
    // VERY VERY DANGEROUS!!! FOR TESTING LOCAL ONLY IN DDEV
    //curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, FALSE);
    //curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE);

and with those opts, it returns invalid client.
Postman with same credentials does get a good oauth token returned.
:)

**Comment by rfay** [comment]
You will have to add the CA or the cert on machine 1. This is explained in https://github.com/FiloSottile/mkcert#installing-the-ca-on-other-systems

**Reply by vpermits** [reply]
Thanks Randy!

**Reply by vpermits** [reply]
Oh! I'm new to containers and DDEV and my description was bad... When I said Machine 1, I meant project ipm.ddev.site on my laptop and united-road-1.ddev.site are BOTH on the same laptop, not separate machines. So, do I need top put the cert(s)? in the containers or...? I'm using PHPStorm for all projects now (hosted by platform.sh) and just doing the ddev config to set up the projects.

**Reply by rfay** [reply]
If they're both on the same host, you shouldn't have to do anything except what's in https://ddev.readthedocs.io/en/latest/users/usage/faq/#can-different-projects-communicate-with-each-other - but I do think you might need the 

```
        external_links:
            - "ddev-router:projectb.ddev.site"
```

Without that, project a thinks that `*.ddev.site` resolves to localhost, so it tries to talk to itself.

**Comment by vpermits** [comment]
Fantastic. I missed that in the FAQ.
BTW, I attended a Drupal Camp CO at the University (I'm in Westminster)
before that lovely COVID shut everything down and heard you speak at a
DDEV session. Really appreciate what you've done for a large community of
users.
Thanks again,
Scott
